src/appleseed/renderer/modeling/bsdf/diffusebtdf.cpp

```diff
@@ -164,8 +166,10 @@ namespace
             if (cos_in * cos_on < 0.0f)
             {
                 // Compute the BRDF value.
-                value = values->m_transmittance;
-                value *= values->m_transmittance_multiplier * RcpPi<float>();
+                value.set(0.0f);
```

Comment:
I'm not sure. The problem with initializing things on demand is that if we add any member to shading components we need to remember to modify all the BSDFs.
In most cases with our current BSDFs, we'll save only one Spectrum::set().
As an alternative we could maybe use a bitfield to mark used components?